docs/release-info/release-notes.md

```diff
@@ -57,63 +57,386 @@ For tips about how to write a good release note, see [Release notes](https://git
+- Added `segment/schemaCache/refreshSkipped/count` metric that shows the number of segments that skipped schema refresh [#18561](https://github.com/apache/druid/pull/18561)
+- Added a metric for when auto-scaling /gets skipped when the publishing task set is not empty [#18536](https://github.com/apache/druid/pull/18536)
+- Added Kafka consumer metrics for indexing tasks in the `prometheus-emitter` default metrics config [#18458](https://github.com/apache/druid/pull/18458)
+- Added an event when a datasource gets dropped from the Coordinator: `segment/schemaCache/datasource/dropped/count` [#18497](https://github.com/apache/druid/pull/18497)
+- Added the `taskType` dimension to the following Prometheus metrics to distinguish between realtime and compaction tasks: `ingest/count`, `ingest/segments/count`, and `ingest/tombstones/count` [#18452](https://github.com/apache/druid/pull/18452)
+- Added a server dimension to segment assignment skip metrics like `segment/assignSkipped/count` [#18313](https://github.com/apache/druid/pull/18313)
+- Changed `CgroupUtil` to be less noisy [#18472](https://github.com/apache/druid/pull/18472)
+- Reduced the clutter of output logs when there's a mismatch in Kafka and task partition set [#18234](https://github.com/apache/druid/pull/18234)
+
 ### Extensions
 
-### Documentation improvements
+#### Iceberg
+
+You can now filter Iceberg data based on a configurable time window.
+
+[#18531](https://github.com/apache/druid/pull/18531)
+
+#### Kubernetes
+
+#### HTTP client
+
+You can now configure what HTTP client library the `Fabric8` `KubernetesClient` uses under the hood to communicate with the k8s server tasks run on. The default remains the same client and config that Druid 34 uses. The additional options that are now available are `okhttp` and a native JDK HttpClient. The default client and config should suffice for most use cases.
+
+Set the `druid.indexer.runner.k8sAndWorker.http.httpClientType` config to `vertx` (default), `okhttp`, or `jdk`.
+
+[#18540](https://github.com/apache/druid/pull/18540)
+
+#### Pod logs
+
+You can now set a timeout for asynchronous operations that interact with k8s pod logs: `druid.indexer.runner.podLogOperationTimeout`.
+
+[#18587](https://github.com/apache/druid/pull/18587)
+
+##### Task logs
+
+For Middle Manager-less ingestion, persisting task logs no longer have the potential to block indefinitely. Instead, there is a time limit for persisting logs, that if breached, results in giving up on the persist. The default timeout is 5 minutes, but can be configured by overriding `druid.indexer.runner.logSaveTimeout` with a duration, such as `PT60S`.
+
+Additionally, task log storage for the Kubernetes runner is now optional. 
+
+[#18444](https://github.com/apache/druid/pull/18444) [#18341](https://github.com/apache/druid/pull/18341)
+
+##### Improved fault tolerance
+
+The Kubernetes extension is now more resilient against certain transient faults. 
```

Comment:
```suggestion
The `druid-kubernetes-overlord-extensions` that uses Kubernetes for launching and managing tasks instead of the Middle Managers is now more resilient against certain transient faults.
```